core/types/receipt.go

```diff
@@ -88,13 +92,29 @@ type receiptRLP struct {
 	CumulativeGasUsed uint64
 	Bloom             Bloom
 	Logs              []*Log
+	FeeInFeeCurrency  *big.Int
+}
+
+type preCIP66receiptRLP struct {
```

Comment:
Maybe we should keep the old structures untouched for easier maintenance and use a specifically-typed structure for the new TX type.

Reply:
The receipt types are a bit confusing, technically here the old ones are untouched, and this one is only used for CELO denominated txs, as seen in the Receipts encode index (and decode as well).

But it's true that it feels like a weird change from this perspective